microsoft/testsuites/core/cpu.py

```diff
@@ -91,18 +91,27 @@ def cpu_count_check(self, node: Node, log: Logger) -> None:
         # 1. Get vCPU count.
         cpu_count = lscpu.get_core_count()
         log.debug(f"{cpu_count} CPU cores detected...")
-        # 2. Caculate vCPU count by core_per_socket_count * socket_count *
-        #  thread_per_core_count.
-        caculated_cpu_count = (
-            lscpu.get_core_per_socket_count()
-            * lscpu.get_socket_count()
-            * lscpu.get_thread_per_core_count()
-        )
+        # 2. Calculate  vCPU count
+        # If there is cluster topology, calculate vCPU count by core_per_cluster_count *
+        # cluster_count * thread_per_core_count, else by core_per_socket_count *
+        # socket_count * thread_per_core_count.
+        if lscpu.check_cluster_exists():
```

Comment:
move this calculation to lscpu, it's useful for other test cases too.

Reply:
Done